frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -144,6 +148,27 @@ local footerTextGeneratorMap = {
             end
         end
     end,
+    frontlight_warmth = function(footer)
+        local symbol_type = footer.settings.item_prefix
+        local prefix = symbol_prefix[symbol_type].frontlight_warmth
+        local powerd = Device:getPowerDevice()
+        if powerd:isFrontlightOn() then
+            if Device:isCervantes() or Device:isKobo() then
+                return (prefix .. " %d"):format(powerd:getWarmth())
+            else
+                local warmth = math.floor(powerd:getWarmth()/100*powerd.fl_warmth_max)
```

Comment:
Does the "stub" variant of `getWarmth` return 0? because if it returns `nil`, this'll probably crash ;).

Reply:
The same applies to the Kobo branch, actually ;).

Reply:
Also, it's already a percent on Kobo/Cervantes, so, it probably ought to be displayed as such, like the FrontLight level ;). (i.e., `%d%%`).

Reply:
Actually, I'd *definitely* make sure the `getWarmth` API is consistent and *always* returns a percentage: i.e., move the Android scaling to the Android powerd ;).

(c.., the get*HW vs. get* stuff powerd is riddled with for this kind of stuff).

Reply:
> The same applies to the Kobo branch, actually ;).
> 
> (Because there are a number of Kobo w/o NaturalLight where it'll return nil ;).)

It seems something is wrong here with the displayed code (maybe caused by my `push force`): If you look under files changed you can see, that I have added a protector in Line 159
```
 elseif powerd:getWarmth() then
```

Reply:
> Actually, I'd _definitely_ make sure the `getWarmth` API is consistent and _always_ returns a percentage: i.e., move the Android scaling to the Android powerd ;).
> 
> (c.., the get_HW vs. get_ stuff powerd is riddled with for this kind of stuff).

That is already done. I will change the display in `devicelistener` to percent, to be consistent.